src/probe/probe.c

```diff
@@ -518,6 +524,57 @@ static int copy_to_pbuffer(struct probe_dma_buf *pbuf, void *data, uint32_t byte
 	return 0;
 }
 
+/**
+ * \brief Copy data from probe buffer and update buffer pointers.
+ * \param[out] probe DMA buffer.
+ * \param[out] data pointer.
+ * \param[in] size.
+ * \return 0 on success, error code otherwise.
+ */
+static int copy_from_pbuffer(struct probe_dma_buf *pbuf, void *data, uint32_t bytes)
+{
+	uint32_t head;
+	uint32_t tail;
+
+	if (bytes == 0)
+		return 0;
+	/* not enough data available so set it to 0 */
+	if (pbuf->avail < bytes) {
+		memset(data, 0, bytes);
+		return 0;
+	}
+
+	if (pbuf->end_addr - pbuf->r_ptr < bytes) {
```

Comment:
The rest of this function looks complex and is completely undocumented.

Reply:
just standard checks for circular buffer and memcpy, comments added